### PR TITLE
Set web-ci-pipeline.yml only triggered when related fields are updated

### PR DESCRIPTION
**Pull request by jchen351**
- 'js/web'
    - 'js/node'
    - 'onnxruntime/core/providers/js'
    is updated

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/web-ci-pipeline.yml [hunk 1]
```diff
@@ -1,3 +1,23 @@
+trigger:
+  branches:
+    include:
+    - main
+    - rel-*
+  paths:
+    include:
+    - 'js/web'
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/web-ci-pipeline.yml:8]: I think it is not enough. The web pipeline depends on the core runtime. So, you cannot just only trigger this pipeline when js file changes. What you can do is: not trigger this pipeline if a PR only contains some unrelated changes such as DNNL EP/TensorRT/...

File: tools/ci_build/github/azure-pipelines/web-ci-pipeline.yml [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+    - 'onnxruntime/core/providers/tensorrt'
+    - 'onnxruntime/core/providers/cuda'
+    - 'onnxruntime/core/providers/rocm'
+pr:
+  branches:
+    include:
+    - main
+    - rel-*
+  paths:
+    include:
```
Comment (fs-eire) [inline tools/ci_build/github/azure-pipelines/web-ci-pipeline.yml:21]: This is too aggresive. a lot of changes should also be caught by web CI:
submodules
cmake
include
tools

**Comment by fs-eire** [comment]
js/node is currently being tested in each CPU CI pipeline (win/x64, win/arm64, linux/x64, linux/arm64, osx/x64, osx/arm64)

and folder js/node does not change frequently. should not exclude them from the CPU pipeline

File: tools/ci_build/github/azure-pipelines/web-ci-pipeline.yml [hunk 3]
```diff
@@ -1,3 +1,45 @@
...
+    - 'include'
+    - 'tool'
+    - '.gitmodules'
+    - 'setup.py'
+    - 'build.*'
+    - '.github'
+    exclude:
+    - 'onnxruntime/core/providers/tensorrt'
+    - 'onnxruntime/core/providers/cuda'
+    - 'onnxruntime/core/providers/rocm'
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/web-ci-pipeline.yml:42]: Please add a task that will delete the unwanted files at the beginning of the build jobs.
The inclusion/exclusion rule here is very complicated. Without some validation, it could soon go out of control.

**Review by snnn — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/web-ci-pipeline.yml (hunk 3 above)
Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/web-ci-pipeline.yml]: Will create a separate PR for web-ci-pipeline.yml after clean out files
